p2panda-rs/src/materialiser/marshall_entries.rs

```diff
@@ -0,0 +1,91 @@
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+use crate::entry::{decode_entry, EntrySigned};
+use crate::materialiser::{Edge, MaterialisationError};
+use crate::message::MessageEncoded;
+
+/// Method for marshalling an array of (EntrySigned, MessageEncoded) into an array of Edges which can then
+/// be turned into a DAG. It is here to detach the DAG from any concepts of Entries and Messages. The DAG just sees
+/// Strings. This step will look different once have some more instance related fields in our Message struct.
+pub fn marshall_entries(
+    entries: Vec<(EntrySigned, MessageEncoded)>,
+) -> Result<Vec<Edge>, MaterialisationError> {
+    let mut edges = Vec::new();
+    for (entry_signed, message_encoded) in entries {
+        let entry = match decode_entry(&entry_signed, Some(&message_encoded)) {
+            Ok(entry) => Ok(entry),
+            Err(err) => Err(MaterialisationError::EntrySignedError(err)),
+        }?;
+
+        if entry.message().is_none() {
+            // The message has been deleted.
+            continue;
+        }
+
+        // If we have a `link` field in the Message then we can use it here.
```

Comment:
I think we can't have that in create messages can we? Because we need to make the message, then make an entry from it, and only then we have the id. It's a chicken and egg situation.

If I have this mixed up don't mind this comment, otherwise this comment can be removed I think.

Reply:
Yes, I think that line may have been left over after some code was removed, not sure what it's alluding to... Thanks! Removed it.